### PR TITLE
WebRTC: Add a test related to https://github.com/w3c/webrtc-pc/issues/2215

### PR DESCRIPTION
**Pull request by youennf**

### CODE REVIEW

File: webrtc/RTCDataChannel-send-blob-order.html [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    channel2.binaryType = "arraybuffer";
+    channel2.onmessage = t.step_func((e) => {
+        if (firstMessage) {
+            assert_equals(e.data.byteLength, data1Size);
+            firstMessage = false;
+            return;
+        }
+        assert_equals(e.data.byteLength, data2Size);
+        done();
```
Comment (jan-ivar) [inline webrtc/RTCDataChannel-send-blob-order.html]: This uses the deferred pattern, which we can avoid it with async/await. E.g.:
```js
    channel1.send(data1);
    channel1.send(data2);
    let e = await new Promise(r => channel2.onmessage = r);
    assert_equals(e.data.byteLength, data1Size);
    e = await new Promise(r => channel2.onmessage = r);
    assert_equals(e.data.byteLength, data2Size);
```
(I've [tested](https://jsfiddle.net/jib1/v2ctmg5L/) that it works) This also gets rid of needing the `t.step_func` function in a `promise_test`.

**Review by jan-ivar — APPROVED** [review]
Thanks! I've filed [bug 1577830](https://bugzilla.mozilla.org/show_bug.cgi?id=1577830) to fix this in Firefox.

File: webrtc/RTCDataChannel-send-blob-order.html (hunk 1 above)
Comment (guest271314) [reply webrtc/RTCDataChannel-send-blob-order.html]: @jan-ivar See https://github.com/w3c/webrtc-pc/issues/2215#issuecomment-526714892

File: webrtc/RTCDataChannel-send-blob-order.html [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+<script src="/resources/testharnessreport.js"></script>
+<script src="RTCPeerConnection-helper.js"></script>
+<script>
+promise_test(async t => {
+    [channel1, channel2] = await createDataChannelPair();
+
+    let done;
+    const promise = new Promise(resolve => done = resolve);
+
+    const data1 = new Blob(['blob']);
```
[the remark this answers is not in the document]
Comment (youennf) [reply webrtc/RTCDataChannel-send-blob-order.html]: ok

File: webrtc/RTCDataChannel-send-blob-order.html (hunk 1 above)
Comment (youennf) [reply webrtc/RTCDataChannel-send-blob-order.html]: nice, updated accordingly

**Pull request merged**